src/services/package-manager/package-service.ts

```diff
@@ -141,121 +140,121 @@ class PackageService {
+        // }
     }
 
-    private async fillNodeDependencies(nodesListToExport: BatchExportNodeTransport[], allPackages: ContentNodeTransport[], actionFlowPackageKeys: string[], nodePath: string[]): Promise<BatchExportNodeTransport[]> {
-        let nodesListWithActiveVersion = await this.getNodesWithActiveVersion(nodesListToExport);
-
+    private async getDependencyPackages(nodesToResolve: BatchExportNodeTransport[], dependencyPackages: BatchExportNodeTransport[], allPackages: ContentNodeTransport[], actionFlowPackageKeys: string[], resolvedDependencies: string[], versionsByNodeKey: Map<string, string[]>): Promise<BatchExportNodeTransport[]> {
         const draftIdByNodeId = new Map<string, string>();
-        nodesListWithActiveVersion.forEach(node => draftIdByNodeId.set(node.workingDraftId, node.id));
+        nodesToResolve.forEach(node => draftIdByNodeId.set(node.activatedDraftId, node.id));
 
-        const dependenciesByPackageIds = await this.getPackagesWithDependencies(draftIdByNodeId, actionFlowPackageKeys);
+        const dependenciesByPackageDraftIds = await this.getPackagesWithDependencies(draftIdByNodeId, actionFlowPackageKeys);
 
-        nodesListWithActiveVersion = nodesListWithActiveVersion.map(nodeToExport => {
-            nodeToExport.dependencies = dependenciesByPackageIds[nodeToExport.workingDraftId] ?? [];
+        const nodesWithDependencies = nodesToResolve.map(nodeToExport => {
+            nodeToExport.dependencies = dependenciesByPackageDraftIds[nodeToExport.activatedDraftId] ?? [];
             return nodeToExport;
-        })
-
-        const variableAssignments = await variableService.getVariableAssignmentsForNodes(nodesListWithActiveVersion);
-
-        nodesListWithActiveVersion.forEach(node => {
-            node.variables = variableAssignments.find(nodeWithVariablesAssignment => nodeWithVariablesAssignment.key === node.key)?.variableAssignments;
         });
 
-        const dataModelAssignments = await dataModelService.getDatamodelsForNodes(nodesListToExport);
-        nodesListWithActiveVersion.forEach(node => {
-            node.datamodels = dataModelAssignments.get(node.key);
-        })
+        for (const node of nodesWithDependencies) {
+            node.dependencies.forEach(dependency => {
+                const dependencyVersions = versionsByNodeKey.get(dependency.key) ?? [];
+                if (!dependencyVersions.includes(dependency.version)) {
+                    dependencyVersions.push(dependency.version);
+                    versionsByNodeKey.set(dependency.key, dependencyVersions);
+                }
+            });
 
-        nodesListToExport = await this.getNodeDependencies(nodesListToExport, allPackages, actionFlowPackageKeys, nodePath);
-        return nodesListToExport
-    }
+            const nodesToGetKeys = node.dependencies.filter(dependency => {
+                return !this.nodeHasBeenResolvedBefore(resolvedDependencies, dependency.key, dependency.version)
```

Comment:
Did we decide that we are safe from circular dependency and do not need to add a check that kills the process